### ISSUE TITLE
panic: unaligned 64-bit atomic operation

### ISSUE THREAD

**Issue by new0nebit** [SiaFoundation/hostd#86]
### Current Behavior

I encountered a runtime panic when running the hostd Docker container on my Raspberry Pi (aarch64).

Here's a little snippet of the log:

```
goroutine 51 [running]:
runtime/internal/atomic.panicUnaligned()
	runtime/internal/atomic/unaligned.go:8 +0x24
runtime/internal/atomic.Store64(0x2c9010c, 0x3e)
	runtime/internal/atomic/atomic_arm.s:291 +0x14
go.sia.tech/hostd/wallet.(*SingleAddressWallet).ProcessConsensusChange(0x2c900e0, {{0xc4, 0x5, 0x69, 0x84, 0xbf, 0x96, 0x84, 0x32, 0xaf, ...}, ...})
	go.sia.tech/hostd/wallet/wallet.go:698 +0xba4
go.sia.tech/siad/modules/consensus.(*ConsensusSet).managedInitializeSubscribe.func2(0x2d28180)
	go.sia.tech/siad@v1.5.10-0.20230228235644-3059c0b930ca/modules/consensus/subscribe.go:212 +0x19c
gitlab.com/NebulousLabs/bolt.(*DB).View(0x29003c0, 0x284ad2c)
	gitlab.com/NebulousLabs/bolt@v1.4.4/db.go:730 +0xa4
go.sia.tech/siad/modules/consensus.(*ConsensusSet).managedInitializeSubscribe(0x28c8a80, {0x10e9308, 0x2c900e0}, {0xef, 0xea, 0xf7, 0xe9, 0x3b, 0x5f, 0xf4, ...}, ...)
	go.sia.tech/siad@v1.5.10-0.20230228235644-3059c0b930ca/modules/consensus/subscribe.go:200 +0x33c
go.sia.tech/siad/modules/consensus.(*ConsensusSet).ConsensusSetSubscribe(0x28c8a80, {0x10e9308, 0x2c900e0}, {0xef, 0xea, 0xf7, 0xe9, 0x3b, 0x5f, 0xf4, ...}, ...)
	go.sia.tech/siad@v1.5.10-0.20230228235644-3059c0b930ca/modules/consensus/subscribe.go:256 +0xec
go.sia.tech/hostd/chain.(*Manager).Subscribe(0x2dba000, {0x10e9308, 0x2c900e0}, {0xef, 0xea, 0xf7, 0xe9, 0x3b, 0x5f, 0xf4, ...}, ...)
	go.sia.tech/hostd/chain/manager.go:133 +0x5c
go.sia.tech/hostd/wallet.NewSingleAddressWallet.func1()
	go.sia.tech/hostd/wallet/wallet.go:767 +0x7c
created by go.sia.tech/hostd/wallet.NewSingleAddressWallet
	go.sia.tech/hostd/wallet/wallet.go:765 +0x720
2023/06/18 11:15:34 hostd v0.1.0-alpha.3-32-g0be24c2
2023/06/18 11:15:34 Network Zen Testnet
{"level":"info","ts":1687086934.2192411,"logger":"stdlog","caller":"hostd/main.go:50","msg":"Using HOSTD_ZEN_API_PASSWORD environment variable."}
{"level":"info","ts":1687086934.220021,"logger":"stdlog","caller":"hostd/main.go:69","msg":"Using HOSTD_ZEN_SEED environment variable."}
panic: unaligned 64-bit atomic operation
```

### Expected Behavior

Work? 📈 😅

### Steps to Reproduce

Just build from the scratch and try to start the container 🤷‍♂️

```
git clone https://github.com/SiaFoundation/hostd.git && cd hostd

docker build -t zen_hostd -f docker/Dockerfile.testnet .  

docker run -d --name hostd -e HOSTD_ZEN_API_PASSWORD='PASS' -e HOSTD_ZEN_SEED='SEED' -p localhost:9880:9880/tcp -p :9881:9881/tcp zen_hostd
```

### Version

hostd v0.1.0-alpha.3-32-g0be24c2

### What operating system did the problem occur on (e.g. Ubuntu 22.04, macOS 12.0, Windows 11)?

Raspberry Pi OS | aarch64

**Comment by n8maninger** [comment]
I can't recreate this on a fresh install of Raspberry Pi OS (64-bit). Are you sure you installed the 64-bit version? Looking at Raspberry Pi Imager, the default option is 32-bit.

![Screenshot 2023-06-19 at 10 45 16 AM](https://github.com/SiaFoundation/hostd/assets/5544150/f35c9dcf-1f35-4d5e-80e5-70e26f29dad1)

**Comment by new0nebit** [comment]
Yeap, absolutely. In addition, the same Raspberry Pi is also carrying a renterd container, which works like clockwork.
I'm very impressed that you managed to do that. I spent a few days trying to exclude all the reasons why this might be happening. I can see the container starting up and hostd launching for about 5 seconds and if I take action quickly I'll be lucky enough to even be able to log in and enter the UI but then it crash. 

<img width="282" alt="Screenshot 2023-06-19 в 19 58 04" src="https://github.com/SiaFoundation/hostd/assets/17514297/d49f818d-3c78-4574-8c65-ddc90b339c19">

For the sake of the experiment's purity, I even tried running hostd on mainnet, although I understood that the result would likely be the same, and indeed it was.

```
2023/06/18 12:56:53 hostd v0.1.0-alpha.3-32-g0be24c2
2023/06/18 12:56:53 Network Mainnet
{"level":"info","ts":1687093013.470032,"logger":"stdlog","caller":"hostd/main.go:50","msg":"Using HOSTD_API_PASSWORD environment variable."}
{"level":"info","ts":1687093013.4703264,"logger":"stdlog","caller":"hostd/main.go:69","msg":"Using HOSTD_SEED environment variable."}
panic: unaligned 64-bit atomic operation

goroutine 69 [running]:
runtime/internal/atomic.panicUnaligned()
	runtime/internal/atomic/unaligned.go:8 +0x24
runtime/internal/atomic.Store64(0x350e56c, 0x0)
	runtime/internal/atomic/atomic_arm.s:291 +0x14
go.sia.tech/hostd/wallet.(*SingleAddressWallet).ProcessConsensusChange(0x350e540, {{0x89, 0xea, 0xd7, 0xf6, 0x71, 0x80, 0xa2, 0x8f, 0x5b, ...}, ...})
	go.sia.tech/hostd/wallet/wallet.go:698 +0xba4
go.sia.tech/siad/modules/consensus.(*ConsensusSet).managedInitializeSubscribe.func2(0x30c6600)
	go.sia.tech/siad@v1.5.10-0.20230228235644-3059c0b930ca/modules/consensus/subscribe.go:212 +0x19c
gitlab.com/NebulousLabs/bolt.(*DB).View(0x31143c0, 0x30f3d2c)
	gitlab.com/NebulousLabs/bolt@v1.4.4/db.go:730 +0xa4
go.sia.tech/siad/modules/consensus.(*ConsensusSet).managedInitializeSubscribe(0x30d0a80, {0x10edae8, 0x350e540}, {0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...)
	go.sia.tech/siad@v1.5.10-0.20230228235644-3059c0b930ca/modules/consensus/subscribe.go:200 +0x33c
go.sia.tech/siad/modules/consensus.(*ConsensusSet).ConsensusSetSubscribe(0x30d0a80, {0x10edae8, 0x350e540}, {0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...)
	go.sia.tech/siad@v1.5.10-0.20230228235644-3059c0b930ca/modules/consensus/subscribe.go:256 +0xec
go.sia.tech/hostd/chain.(*Manager).Subscribe(0x3194000, {0x10edae8, 0x350e540}, {0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...)
	go.sia.tech/hostd/chain/manager.go:133 +0x5c
go.sia.tech/hostd/wallet.NewSingleAddressWallet.func1()
	go.sia.tech/hostd/wallet/wallet.go:767 +0x7c
created by go.sia.tech/hostd/wallet.NewSingleAddressWallet
	go.sia.tech/hostd/wallet/wallet.go:765 +0x720

```

**Comment by n8maninger** [comment]
What’s the output of `sudo getconf LONG_BIT`. Did you build the renterd container manually or pull the existing one?

**Comment by new0nebit** [comment]
<img width="266" alt="Screenshot 2023-06-20 в 04 42 14" src="https://github.com/SiaFoundation/hostd/assets/17514297/f26ef78c-e17c-42eb-ba3d-12a307936599">

Manually. When I first tried to pull the existing one, it also did not work. And by the way, if I'm not mistaken, the same thing happened with renterd. And only when I did it manually, renterd started to work. It has been working for about a month now, I update and use it almost every day and everything works fine.

**Comment by n8maninger** [comment]
Very odd. I can fix the alignment, but I can't recreate this issue on a fresh install of Raspberry Pi OS (64-bit). I can build the container, start it, fully sync, and access the UI. Does pulling the `ghcr.io/siafoundation/hostd:master` image result in the same panic or a different one?

**Comment by new0nebit** [comment]
<img width="567" alt="Screenshot 2023-06-20 в 05 17 42" src="https://github.com/SiaFoundation/hostd/assets/17514297/89414d24-b408-4c7e-84b0-5870c226df08">

Okay, here it is. Now I remember what was the error for renterd. That's exactly why I do this manually. Having seen that the renterd container has been working fine for a long period of time, I decided to do the same trick with hostd, but you already know the result. Can you please tell me, are you able to pull the ghcr.io/siafoundation/hostd:master image? My pupils will dilate even more if you confirm this 😵‍💫

And YES. Binary works just fine 👌

**Comment by n8maninger** [comment]
> Can you please tell me, did you build it from scratch or you able to pull the `ghcr.io/siafoundation/hostd:master`

I can build it from scratch or pull the image. How did you install Docker? It seems like your Docker is trying to pull the 32-bit image, which we don't support.

**Comment by n8maninger** [comment]
#89 will fix the crash. However, I'd still recommend that you figure out why your Docker install is trying to grab the 32-bit image at some point.

**Comment by new0nebit** [comment]
That's exactly what I'm doing right now 🫡 Your question about Docker put me in the mindset that I had installed the wrong version of Docker and I managed to find this command `docker version --format '{{.Server.Arch}}'

<img width="442" alt="Screenshot 2023-06-20 в 06 36 12" src="https://github.com/SiaFoundation/hostd/assets/17514297/ecde47b8-82ac-419a-a5b9-ebe3c0155932">

If I'm right, for a 64-bit Docker, the architecture returned will be `amd64` or `aarch64`. But for a 32-bit installation, it will  be: `386` or `arm`. Presently, I will engage in the task of uninstalling and reinstalling Docker. I will be boundlessly joyful if this succeeds, yet concurrently saddened by my own lack of attentiveness 😔
